ckanext/xloader/jobs.py

```diff
@@ -336,17 +336,18 @@ def _download_resource_data(resource, data, logger):
     url_parts = urlsplit(url)
     scheme = url_parts.scheme
 
-    # check if it is an uploaded file
-    domain = url_parts.netloc
-    site_url = config.get('ckan.site_url')
-    if resource.get('url_type') != 'upload' and domain != site_url:
-        raise JobError('Only uploaded files can be added to the Data Store.')
+    # (canada fork only): uploaded and white listed only
+    allowed_domains = config.get('ckanext.canada.datastore_source_domain_white_list', '').split()
```

Comment:
we've got config_declaration.yaml support now, so we could declare this as a list instead of handling parsing here

Reply:
also, `allow_list` instead? I heard white list is problematic

Reply:
using config declaration from canad aplugin now, type is list.

and refactored the three plugins to `allow_list`